src/common/str_list.cc

```diff
@@ -18,89 +18,67 @@ using std::string;
 using std::vector;
 using std::set;
 using std::list;
-
-static bool get_next_token(const string &s, size_t& pos, const char *delims, string& token)
-{
-  int start = s.find_first_not_of(delims, pos);
-  int end;
-
-  if (start < 0){
-    pos = s.size();
-    return false;
-  }
-
-  end = s.find_first_of(delims, start);
-  if (end >= 0)
-    pos = end + 1;
-  else {
-    pos = end = s.size();
-  }
-
-  token = s.substr(start, end - start);
-  return true;
-}
+using ceph::for_each_substr;
 
 void get_str_list(const string& str, const char *delims, list<string>& str_list)
 {
-  size_t pos = 0;
-  string token;
-
   str_list.clear();
-
-  while (pos < str.size()) {
-    if (get_next_token(str, pos, delims, token)) {
-      if (token.size() > 0) {
-        str_list.push_back(token);
-      }
-    }
-  }
+  for_each_substr(str, delims, [&str_list] (boost::string_view token) {
```

Comment:
string_view is just a pointer/len pair, so it's trivial to copy and can be passed in registers. some advice about this from the [cpp core guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in):

> F.16: For “in” parameters, pass cheaply-copied types by value and others by reference to const
> ...
> What is “cheap to copy” depends on the machine architecture, but two or three words (doubles, pointers, references) are usually best passed by value. When copying is cheap, nothing beats the simplicity and safety of copying, and for small objects (up to two or three words) it is also faster than passing by reference because it does not require an extra indirection to access from the function.